### ISSUE TITLE
[FR] Straightforward way of sending additional querystring parameters to the backend

### ISSUE THREAD

**Issue by andreafdaf** [marmelab/react-admin#7219]
Hi! I'm using RA and really loving it but there is something that make it "difficult" to integrate into existing backends (we are migrating a multi hundred endpoint application to RA).

Something we're missing is a straightforward possibility of sending custom query parameters from the components/hooks to the dataProvider and then from the dataProvider to the backend. What I would like to propose is a really simple mechanism to do it, like:
```jsx
<List queryParams={{ populate: true }} {...props}>
```
```jsx
<Create queryParams={{ template: 'sales' }} {...props}>
```
```jsx
<Edit queryParams={{ reason: 'correction' }} {...props}>
```
And consequently (just one to make my point):
```ts
useGetList(
  'customers',
  {
    sort: ...
    filter: ...
    queryParams: { populate: true }
  }
)
```
And for example on ```ra-data-simple-rest```:
```ts
getList: (resource, params) => { // inside params we have 'queryParams'
...
const query = {
    ...(params.queryParams ?? {}), // just a simple hack, it can be done in better ways
    sort: JSON.stringify([field, order]),
    range: JSON.stringify([rangeStart, rangeEnd]),
    filter: JSON.stringify(params.filter),
}
```

Creating custom resources, custom dataProviders, custom methods on the dataProvider... those all look far too complicated and cumbersome to achieve such a simple functionality.

I don't know if this is a breaking change or not, I'm proposing it as a standalone issue since I don't think it is backward incompatible.

Thanks for your time and for this library!

**Comment by WiXSL** [comment]
For current version of react-admin you have a [solution here](https://stackoverflow.com/questions/55713128/react-admin-how-to-pass-more-parameters-to-a-dataprovider-for-type-get-list/55731133#55731133).

In the next mayor version, coming soon, you would be able to send extra parameters to any function of a dataProvider.

So, I'm closing this issue, since as an enhancement request it has already been address for version 4 of react-admin.

**Comment by andreafdaf** [comment]
Hi @WiXSL, thanks for this quick reply.

I don't think the v3 solution would work out since it relies on adding the parameter to filters, not as an autonomous querystring parameter. Anyway we are already building on v4 so to be already compatible once it comes out.

As for the v4 enhancement, are you talking about the 'meta' property? https://github.com/marmelab/react-admin/blob/410971a6eda3347a038ac370c5db0a8449de5eef/packages/ra-core/src/dataProvider/useGetList.ts#L78

It might work but looking for example at https://github.com/marmelab/react-admin/blob/410971a6eda3347a038ac370c5db0a8449de5eef/packages/ra-core/src/controller/list/useListController.ts#L82 it seems that there is no way to pass the meta property from the List component down to the dataProvider.

Am I missing something?

Thanks again!

BTW: if a subscription is necessary to go this deep into the conversation we'll happily purchase one (we already want to, so no problem at all to purchase it right now)

**Comment by WiXSL** [comment]
Yes, I'm talking about the `meta` parameter.
Check out the definition of `useGetList` you can pass as one of the params a meta argument
https://github.com/marmelab/react-admin/blob/410971a6eda3347a038ac370c5db0a8449de5eef/packages/ra-core/src/dataProvider/useGetList.ts#L53

**Comment by djhi** [comment]
> it seems that there is no way to pass the meta property from the List component down to the dataProvider.

Not yet

**Comment by andreafdaf** [comment]
@djhi I guess that my FR boils down to:

-> On v4, possibility of passing the ```meta``` property down from List, Create, Edit to the dataProvider.

BTW thanks for such a quick feedback, we just subscribed to the enterprise edition 🚀

**Comment by fzaninotto** [comment]
For your information, this has been added to react-admin, and released in 4.2.0. See #7841 for details.
